src/canmatrix/canmatrix.py

```diff
@@ -308,6 +309,8 @@ def phys2raw(self, value=None):
                         "{} is invalid value choice for {}".format(value, self)
                 )
 
+        if not isinstance(value, defaultFloatFactory):  # use the same data type as offset and factor
+            value = defaultFloatFactory(value)
```

Comment:
I'm trying to work us away from this.  The caller should pass the proper thing rather than us guessing what the proper thing is.  In particular here, this check doesn't work for a `defaultFloatFactory` that isn't a class.

Reply:
Okay but this works , the defaultFloatFactory  is `decimal.Decimal` so it's a class. But it can become a function one day - I understand.
The problem is, that if we pass an `int` to the function, it produces _a result_, but (slightly) wrong result, instead of raising an exception.

Reply:
I'm admittedly pretty tired right now...  Could you provide an example where the original code got an integer input wrong?

Reply:
When being tired by my faulty PR, don't hesitate to say it :-D

I wrote nonsense. Regarding phys2raw the `int` works and `float` raises `TypeError('unsupported operand type(s) for -:')`. Maybe try/catch + raise with better explanation. Or simply let it be.

Reply:
Interesting, so it does.  Yeah, I think I'd leave it for now.  It tells you the line and the two incompatible types.  Either the caller just forgot or they probably ought to learn about the issues with floats.  *shrug*

On the reraise topic, in py3 be sure to `raise DescriptiveError() from original_exception` so the original isn't lost.  Sadly there's no py2/3 compatible solution that isn't totally custom so we don't get to do that.  Leaves me more hesitant to make 'helpful' replacement exceptions.